e2e-tests/IIoTPlatform-E2E-Tests/Standalone/A_PublishSingleNodeStandaloneDirectMethodTestTheory.cs

```diff
@@ -138,6 +138,18 @@ IIoTMultipleNodesTestContext context
             Assert.Equal(jsonResponse.Count, 1);
             Assert.Equal(jsonResponse[0].Id, "nsu=http://microsoft.com/Opc/OpcPlc/;s=SlowUInt1");
 
+            //Call GetDiagnosticInfo direct method
+            var responseGetDiagnosticInfo = await CallMethodAsync(
+                new MethodParameterModel {
+                    Name = TestConstants.DirectMethodNames.GetDiagnosticInfo,
+                },
+                cts.Token
+            ).ConfigureAwait(false);
+
+            Assert.Equal((int)HttpStatusCode.OK, responseGetDiagnosticInfo.Status);
+            var jsonResponseGetDiagnosticInfo = _serializer.Deserialize<List<JobDiagnosticInfoModel>>(responseGetDiagnosticInfo.JsonPayload);
+            Assert.Equal(jsonResponseGetDiagnosticInfo.Count, 1);
```

Comment:
Please also validate that fields have proper values. I think that we can have the following checks for an active processing engine:

diagInfo.IngressValueChanges > 0
diagInfo.IngressDataChanges > 0
diagInfo.MonitoredOpcNodesFailedCount == 0
diagInfo.MonitoredOpcNodesSucceededCount == (number of nodes that we subscribed to)
diagInfo.OpcEndpointConnected == true
diagInfo.OutgressIoTMessageCount > 0
diagInfo.EncoderNotificationsDropped == 0
diagInfo.OutgressInputBufferDropped == 0

We can have a method that checks all of those and call it for all diagnostic info objects that we get back from a direct method.

Reply:
I am trying to do something similar in #1563, please have a look.

Reply:
added